mouseion/mouseion/tasks.py

```diff
@@ -0,0 +1,196 @@
+        self.tasks: List[Task] = []  # Tasks that are yet to be started
+        self.completed_tasks: Set[int] = set()
+        self.console.print(f'[aw.c]Artificial Wisdom™ NLP Tools [/aw.c][aw.b]•[/aw.b] [aw.c]{command}[/aw.c]')
+
+    def add_tasks(self, tasks: List[Task]) -> None:
+        self.all_tasks.extend(tasks)
+        self.tasks.extend(tasks)
+
+    def process_tasks(self) -> None:
+        with Progress(
+            TextColumn("•", style="aw.b"),
+            TextColumn("[progress.description]{task.description}"),
+            TextColumn("•", style="aw.b"),
+            BarColumn(style="aw.a", complete_style="aw.b"),
+            TextColumn("•", style="aw.b"),
+            TimeRemainingColumn(),
+            TextColumn("•", style="aw.b"),
+            TimeElapsedColumn(),
+            TextColumn("•", style="aw.b"),
+            TextColumn("[progress.percentage]{task.percentage:>3.0f}%"),
+            TextColumn("•", style="aw.b"),
+            HumanReadableProgress(),
+            TextColumn("•", style="aw.b"),
+            console=self.console,
+        ) as progress:
+            task_progress_map: Dict[int, int] = {task.id: progress.add_task(
+                f'{task.description}',
+                total=task.size,
+                fields={'type': task.progress_type}
+            ) for task in self.all_tasks}
+            processes = []
+            while self.tasks or processes:
+                # Start ready tasks
+                ready_tasks = [t for t in self.tasks if t.is_ready(self.completed_tasks)]
+                for t in ready_tasks:
+                    process = Process(target=t.process_function, args=(t.size, t.queue))
+                    processes.append((t.id, process, t.queue))
+                    process.start()
+                    self.tasks.remove(t)
+
+                # Avoid busy waiting if no tasks are ready
+                if not ready_tasks:
+                    time.sleep(0.1)
+
+                # Update progress and check for completed tasks
+                for task_id, process, queue in processes:
+                    while not queue.empty():
+                        progress_value = queue.get()
+                        progress.update(task_progress_map[task_id], advance=1)
+                        if progress_value >= self.get_task_size(task_id):  # Assuming task is completed when progress reaches size
+                            self.completed_tasks.add(task_id)
+                            progress.update(task_progress_map[task_id], completed=True)
+
+                # Clean up completed processes
+                processes = [(task_id, p, q) for task_id, p, q in processes if p.is_alive()]
+
+    def get_task_size(self, task_id: int) -> int:
+        for task in self.all_tasks:
+            if task.id == task_id:
+                return task.size
+        return 0
+
+def task_load_jsonl(size: int, queue: Queue) -> float:
+    progress = 0
+    for i in range(size):
+        time.sleep(0.3)
```

Comment:
just curious as to why you've embedded a static sleep?

Reply:
this is an example function - the goal was to add a sample to the task handler and have it take 10 seconds.